catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java

```diff
@@ -369,18 +396,36 @@ public Metacard transform(InputStream input) throws IOException, CatalogTransfor
         return transform(input, null);
     }
 
+    protected MetacardType getExtendedMetacardType(String metacardContentType) {
+        MetacardType metacardType = getMetacardTypeFromMimeType(metacardContentType);
+        if (metacardType == null) {
+            metacardType = commonTikaMetacardType;
+        }
+
+        Set<AttributeDescriptor> additionalAttributes = getAllExtractorAttributes();
+
+        // The MetacardTypeImpl constructor throws and exception if the set is empty.
+        if (!additionalAttributes.isEmpty()) {
+            metacardType = new MetacardTypeImpl(metacardType.getName(),
+                    metacardType,
+                    additionalAttributes);
+        }
+
+        return metacardType;
+    }
+
     @Override
     public Metacard transform(InputStream input, String id)
             throws IOException, CatalogTransformerException {
         LOGGER.debug("Transforming input stream using Tika.");
-
+        long bytes;
         if (input == null) {
             throw new CatalogTransformerException("Cannot transform null input.");
         }
 
         try (TemporaryFileBackedOutputStream fileBackedOutputStream = new TemporaryFileBackedOutputStream()) {
             try {
-                IOUtils.copy(input, fileBackedOutputStream);
+                bytes = IOUtils.copyLarge(input, fileBackedOutputStream);
```

Comment:
The `IOUtils.copyLarge` method (which is ultimately called by `IOUtils.copy`) returns the number of bytes in the stream. Computing resource size is part of the requirements for another project, so the bytes is captured here. Later, a new attribute for resource size is created and added to the metacard.